ms-common-auth/src/main/java/net/trajano/ms/authz/internal/TokenCache.java

```diff
@@ -0,0 +1,147 @@
+package net.trajano.ms.authz.internal;
+
+import net.trajano.ms.auth.token.ErrorCodes;
+import net.trajano.ms.auth.token.IdTokenResponse;
+import net.trajano.ms.common.oauth.OAuthTokenResponse;
+import net.trajano.ms.core.CryptoOps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.time.Instant;
+
+@Component
+public class TokenCache {
+
+    /**
+     * Logger for TokenCache
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(TokenCache.class);
+
+    @Value("${token.accessTokenExpiration:300}")
+    private int accessTokenExpirationInSeconds;
+
+    @Autowired
+    private CacheManager cm;
+
+    private Cache refreshTokenToEntry;
+
+    private Cache accessTokenToEntry;
+
+    @Autowired
+    private CryptoOps cryptoOps;
+
+    /**
+     * This will return null if a valid entry was not found.
+     * 
+     * @param accessToken
+     *            access token
+     * @param clientId
+     *            client ID
+     * @return OAuth 2.0 ID Token Response
+     */
+    public IdTokenResponse get(final String accessToken,
+        final String clientId) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "clientId". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)